tools/PyTorch/lstm/PyTorch_lstm.py

```diff
@@ -0,0 +1,84 @@
+import math
+import torch
+
+
+# The LSTM model
+def lstm(weight, bias, hidden, cell, _input):
+    # NOTE this line came from: gates = hcat(input,hidden) * weight .+ bias
+    gates = torch.cat((_input, hidden, _input, hidden)) * weight + bias
+    hsize = hidden.shape[0]
+    forget = torch.sigmoid(gates[0: hsize])
+    ingate = torch.sigmoid(gates[hsize: 2 * hsize])
+    outgate = torch.sigmoid(gates[2 * hsize: 3 * hsize])
+    change = torch.tanh(gates[3 * hsize:])
+    cell = cell * forget + ingate * change
+    hidden = outgate * torch.tanh(cell)
+    return (hidden, cell)
+
+
+# Predict output given an input
+def predict(w, w2, s, x):
+    s2 = torch.tensor(s)
+    # NOTE not sure if this should be element-wise or matrix multiplication
+    x = x * w2[0]
```

Comment:
Actually either is fine.   It should be matrix multiplication, but in some cases that matrix might be diagonal, which turns out to be elementwise.  i.e. in matlab notation 
```
diag(w) * x = w .* x
```